### DISCUSSION TITLE
Is it possible to optional use previous version of next/dynamic in the Next 13.1+?

### DISCUSSION THREAD

**Discussion by Marsony** [vercel/next.js#45208] [Help]
Hello everyone. (appDir: false)

Im using next/dynamic in the root of the application (app.tsx) together with the useEffect which mutates my stores (I want do it only on the client side and only **after full hydration**).

Previously next/dynamic doesn't ignore's my painting react phase after useEffect, but right now it skipping because next/dynamic using lazy && suspense.

Suspense ignore's painting phase after useEffect on the client-side but perform code inside which mutates my stores, so, it leads to hydration errors.

Maybe we have several ways to mutates stores on the client side before suspense will freeze rendering phases? Or maybe **fire useEffect after all suspenses**?

**Comment by icyJoseph** [comment]
I think the root issue is that React 18 can await suspenseful stuff... I sounds like hiding the dynamic component behind a boolean flag is not what you'd need in this case. Could you share some pseudo-code?

**Reply by Marsony** [reply]
> I think the root issue is that React 18 can await suspenseful stuff... I sounds like hiding the dynamic component behind a boolean flag is not what you'd need in this case. Could you share some pseudo-code?

Sure
```
import dynamic from 'next/dynamic';
import { useEffect } from 'react';

// Navbar.tsx
const Navbar = () => {
  const anonymous = useSomeIsAuthStore($someAtomicStoreValue);

  return (
    <>
      {!anonymous && <div>Welcome back!</div>}
      {anonymous && <div>Please log in</div>}
    </>
  );
};

// Footer.tsx
const Footer = () => {
  const anonymous = useSomeStore($someAtomicStoreValue);

  return (
    <>
      {!anonymous && <div>...anonymousLinks</div>}
      {anonymous && <div>..authorizedLinks</div>}
    </>
  );
};

// _app.tsx

const SuspensedNavbar = dynamic(() => import('./Navbar.tsx'), {
  ssr: **true**
})

const SuspensedFooter = dynamic(() => import('./Footer.tsx'), {
  ssr: **true**
})

const App = () => {
  useEffect(() => {
    mutateAllStores();
  }, []);

  return (
    <>
      <SomeSuspensedWrapper>
        <SuspensedNavbar />
        <SuspensedFooter />
      </SomeSuspensedWrapper>
    </>
  );
};
```

With previous version of the next/dynamic all works fine because dynamically loaded component doesn't skipping painting phase after useEffect. Right now useEffect completing inside but skipping panting phase because next/dynamic using Suspense && lazy inside
